src/__internal__/utils/logger/logger.test.ts

```diff
@@ -0,0 +1,22 @@
+import Logger from ".";
+import { expectConsoleOutput } from "../../../__spec_helper__/__internal__/test-utils";
+
+test("should not output a warning to the console when logging is disabled", () => {
+  Logger.setEnabledState(false);
+  const consoleWarnSpy = jest
+    .spyOn(console, "warn")
+    .mockImplementation(() => {});
+  Logger.deprecate("This is a deprecation message");
+
+  expect(consoleWarnSpy).not.toHaveBeenCalled();
+});
+
+test("should output a warning to the console with a deprecation prefix when logging is enabled", () => {
+  Logger.setEnabledState(true);
+  const assert = expectConsoleOutput(
+    "[Deprecation] This is a deprecation message",
+    "warn"
+  );
+  Logger.deprecate("This is a deprecation message");
+  assert();
```

Comment:
**suggestion**: what do you think of mocking the console and checking its been called directly?

This would avoid the linter raising the "no assertions" warning.

```suggestion
test("should output a warning to the console with a deprecation prefix when logging is enabled", () => {
  Logger.setEnabledState(true);
  const consoleWarnSpy = jest.spyOn(console, "warn").mockImplementation(() => {});

  Logger.deprecate("This is a deprecation message");
  
  expect(console.warn).toHaveBeenCalledWith("[Deprecation] This is a deprecation message");
  consoleWarnSpy.mockReset();
```